network/p2p/inspector/validation/control_message_validation.go

```diff
@@ -307,12 +332,55 @@ func (c *ControlMsgValidationInspector) validateTopics(ctrlMsgType p2p.ControlMe
 	return nil
 }
 
-// validateTopic the topic is a valid flow topic/channel.
-// All errors returned from this function can be considered benign.
-func (c *ControlMsgValidationInspector) validateTopic(topic channels.Topic) error {
+// validateTopic ensures the topic is a valid flow topic/channel.
+// Expected error returns during normal operations:
+//   - channels.ErrInvalidTopic: if topic is invalid.
+//
+// This func returns an exception in case of unexpected bug or state corruption if cluster prefixed topic validation
+// fails due to unexpected error returned when getting the active cluster IDS.
+func (c *ControlMsgValidationInspector) validateTopic(topic channels.Topic, ctrlMsgType p2p.ControlMessageType) error {
+	channel, ok := channels.ChannelFromTopic(topic)
+	if !ok {
+		return channels.NewInvalidTopicErr(topic, fmt.Errorf("failed to get channel from topic"))
+	}
+
+	// handle cluster prefixed topics
+	if channels.IsClusterChannel(channel) {
+		return c.validateClusterPrefixedTopic(topic, ctrlMsgType)
```

Comment:
To handle edge cases involving epoch transitions and active cluster ID changes, it is important to differentiate between validation failures due to invalid cluster topics and those related to non-cluster topics. One possible approach is to wrap the error in this situation with an ErrInvalidClusterTopic and wrap other errors returned by the methods as an ErrorInvalidTopic. This way, when the error is passed to [this location](https://github.com/onflow/flow-go/blob/c80854e33a9cd52409d350e192ccbb6901680cad/network/p2p/inspector/validation/control_message_validation.go#L274), the consumer can discern between these two cases.

Since this will significantly impact the scoring mechanism, ensure that you also create unit tests to verify the returned error types and integration tests to check the error types sent to the distributor. You may repurpose [this test](https://github.com/onflow/flow-go/blob/87828553813531328e6d086f96a1c1bb787d5e20/insecure/rpc_inspector/validation_inspector_test.go#L131) to develop the integration testing scenario.